29786 ship light dark (#30244)
mdn/content: files/en-us/web/css/color_value/light-dark/index.md

---
title: light-dark()
slug: Web/CSS/color_value/light-dark
page-type: css-function
browser-compat: css.types.color.light-dark
---

{{CSSRef}}

> The **`light-dark()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) allows a user to set light and dark colors, without using the [`prefers-color-scheme`](/en-US/docs/Web/CSS/@media/prefers-color-scheme) [media feature](/en-US/docs/Web/CSS/CSS_media_queries/Using_media_queries#targeting_media_features) and detects if a user has requested light or dark color theme.

Author: I agree, especially as further I have refered to this specifically as a CSS `<color>` function.
It still hurts me to spell color wrong after all these years.

- `dark-color`
  - : {{CSSXref("&lt;color&gt;")}} value to be set for dark {{CSSXref("color-scheme")}}.

### Formal syntax

{{csssyntax}}

## Example

### Setting the colors based on system preferences

In a supporting browser, the colors will change based upon the user preference through an operating system setting (e.g. light or dark mode) or a user agent setting. You can also change this setting in the browsers developer tools.

#### HTML

```html
<h1><code>light-dark()</code> CSS function</h1>
<section>
  <h2>Automatic</h2>
  <p>This section will react to the users system or user agent setting.</p>
</section>
<section class="light">
  <h2>Light</h2>
  <p>
    This section will be light due to the <code>color-scheme: light;</code>.
  </p>
</section>
<section class="dark">
  <h2>Dark</h2>
  <p>This section will be light due to the <code>color-scheme: dark;</code>.</p>
</section>
```

#### CSS

```css
:root {
  /* this has to be set to switch between light or dark */
  color-scheme: light dark;
  --light-bg: ghostwhite;
  --dark-bg: darkslategray;
  --light-color: darkslategray;
  --dark-color: ghostwhite;
  --light-code: tomato;
  --dark-code: gold;

> }

Reviewer: (in other words, you may like the original better, and if you do, no need to edit. I just took a few seconds too long to parse it, so made the suggestion)

> }

Author: This makes a lot of sense to me also.

Users are able to indicate their color-scheme preference through their operating system settings (e.g. light or dark mode) or their user agent settings. The `light-dark()` function enables providing two color values where any `<color>` value is accepted. The `light-dark()` CSS color function returns the first value if the user's preference is set to `light` or if no preference is set and the second value if the user's preference is set to `dark`.

To enable support for the `light-dark()` color function, the {{CSSXref("color-scheme")}} must have a value of `light dark`, usually set on the {{CSSXref(":root")}} [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes).

```css
:root {
  color-scheme: light dark;

> }

Reviewer: ```suggestion
}
body {
  color: light-dark(#333b3c, #efefec);
  background-color: light-dark(#efedea, #223a2c);
}
```

Reviewer: Adding so that the first code block on the page uses the function that is the topic of the page.

```

## Syntax

```css
/* Named color values */
color: light-dark(white, black);

/* RGB color values */

> color: light-dark(rgb(255 255 255), rgb(0 0 0));

Reviewer: Should these be inverted or should this be set to background color? The light color scheme generally has a light background with dark text.

Author: Yes I believe that you're right.

Author: done

/* Custom properties */
color: light-dark(var(--light), var(--dark));
```

### Values

Functional notation: `light-dark(light-color, dark-color)`

- `light-color`

  - : {{CSSXref("&lt;color&gt;")}} value to be set for light {{CSSXref("color-scheme")}}.

- `dark-color`
  - : {{CSSXref("&lt;color&gt;")}} value to be set for dark {{CSSXref("color-scheme")}}.

### Formal syntax

{{csssyntax}}

## Example

### Setting colors based on color scheme

> In a supporting browser, the colors will change based on the user preference through an operating system setting (e.g., light or dark mode) or a user agent setting. You can also change this setting in the browsers {{glossary("developer tools")}}.

Reviewer: ```suggestion
By default, in supporting browsers, the color returned by the `light-dark()` color function depends on the user preference set through an operating system's settings (e.g., light or dark mode) or from a user agent setting. You can also change this setting in the browser's {{glossary("developer tools")}}.
```

#### HTML

> We include three sections to enable targeting light colors, dark colors, and the user's preferred colors.

Reviewer: ```suggestion
We include three sections to enable targeting light colors, dark colors, and the colors selected based on the user's preferred color scheme.
```

```html
<h1><code>light-dark()</code> CSS function</h1>
<section>
  <h2>Automatic</h2>
  <p>This section will react to the users system or user agent setting.</p>
</section>
<section class="light">
  <h2>Light</h2>
  <p>
    This section will be light due to the <code>color-scheme: light;</code>.
  </p>
</section>
<section class="dark">
  <h2>Dark</h2>
  <p>This section will be light due to the <code>color-scheme: dark;</code>.</p>
</section>
```

#### CSS

We include colors for both light and dark themes. We also define `color-scheme` for the document on the `:root` to enable the `light-dark()` color function for the entire document.

```css-nolint
:root {
  /* this has to be set to switch between light or dark */
  color-scheme: light dark;

  --light-bg: ghostwhite;
  --light-color: darkslategray;
  --light-code: tomato;

  --dark-bg: darkslategray;
  --dark-color: ghostwhite;
  --dark-code: gold;
}
* {
  background-color: light-dark(var(--light-bg), var(--dark-bg));
  color: light-dark(var(--light-color), var(--dark-color));
}
code {
  color: light-dark(var(--light-code), var(--dark-code));
}
```

> The `color-scheme` property enables overriding a user's color scheme for document sections. Forcing a page section to only use a light or dark color scheme can be done by setting the `color-scheme` property to `light` or `dark`.

Reviewer: ```suggestion
In addition to enabling the `light-dark()` function, the `color-scheme` property enables overriding a user's color scheme for document sections. Forcing a page section to only use a light or dark color scheme can be done by setting the `color-scheme` property to `light` or `dark`.
```

Users are able to indicate their color-scheme preference through their operating system settings (e.g. light or dark mode) or their user agent settings. The `light-dark()` function enables providing two color values where any `<color>` value is accepted. The `light-dark()` CSS color function returns the first value if the user's preference is set to `light` or if no preference is set and the second value if the user's preference is set to `dark`.

To enable support for the `light-dark()` color function, the {{CSSXref("color-scheme")}} must have a value of `light dark`, usually set on the {{CSSXref(":root")}} [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes).

```css
:root {
  color-scheme: light dark;
}
body {
  color: light-dark(#333b3c, #efefec);
  background-color: light-dark(#efedea, #223a2c);
}
```

## Syntax

```css
/* Named color values */
color: light-dark(black, white);

/* RGB color values */
color: light-dark(rgb(0 0 0), rgb(255 255 255));

/* Custom properties */

> color: light-dark(var(--dark), var(--light));

Reviewer: ```suggestion
color: light-dark(var(--light), var(--dark));
```

Author: I swapped this when swapping the others thinking that the text on a light background should be dark and vice versa.
But looking now it doesn't seem intuitive.
